dotnet/src/SemanticKernel.Core/Orchestration/KernelExecutionContext.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.Services;
+
+namespace Microsoft.SemanticKernel.Orchestration;
+
+/// <summary>
+/// Kernel execution context implementation
+/// This will live during the execution of functions.
+/// </summary>
+internal sealed class KernelExecutionContext : IKernelExecutionContext, IDisposable
+{
+    private Kernel _kernel;
```

Comment:
Seemed this can be `readonly`

Reply:
Agree, `kernel` will be lazy loaded at some point, but for now readonly works.